doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,832 @@
+
+
+#### 2.3.2 Non-replicated state
+
+###### 2.3.2.1 `messagelog` database: not replicated
+
+The messagelog database is not replicated. Each node has its own separate messagelog database. **However**, in order to
+support PostgreSQL streaming replication (hot standby mode) for the serverconf data, the serverconf and messagelog
+databases must be separated. This requires modifications to the installation (a separate PostgreSQL instance is needed
+for the messagelog database) and has some implications on the security server resource requirements as a separate
+instance uses some memory.
+
+
+###### 2.3.2.2 OCSP responses from `/var/cache/xroad/`: not replicated
+
+The OCSP responses are currently not replicated. Replicating them could make the cluster more fault tolerant but the
+replication cannot simultaneously create a single point of failure. A distributed cache could be used for the responses.
+
+
+## 3. X-Road Installation and configuration
+
+This chapter details the complete installation on a high level, with links to other chapters that go into the details.
+
+You can set up the cluster manually, or use the provided Ansible playbook \[[SS-CLUSTER](#12-references)\] if it suits
+your purposes.
+
+### 3.1 Prerequisites
+
+In order to properly set up the data replication, the slave nodes must be able to connect to:
+* the master server using SSH (tcp port 22), and
+* the master `serverconf` database (e.g. tcp port 5433).
+
+
+### 3.2 Master installation
+
+1. Install the X-Road security server packages using the normal installation procedure or use an existing standalone node.
+2. Stop the xroad services.
+3. Create a separate PostgreSQL instance for the `serverconf` database (see section
+   [4. Database replication setup](#4-database-replication-setup) for details).
+4. Change `/etc/xroad/db.properties` to point to the separate database instance:
+   * `serverconf.hibernate.connection.url` : Change the url port number from `5432` to `5433` (or the port you specified)
+5. If you are using an already configured server as the master, the existing configuration was replicated to the slaves
+   in step 3. Otherwise, proceed to configure the master server: install the configuration anchor, set up basic information,
+   create authentication and signing keys and so on. See the security server installation guide \[[IG-SS](#12-references)\]
+   for help with the basic setup.
+6. Set up the configuration file replication, see section
+   [5. Configuring data replication with rsync over SSH](#5-configuring-data-replication-with-rsync-over-ssh)
+   * Additionally, `rssh` shell can be used to to restrict slave access further, but note that it is not available on RHEL.
+
+7. Configure the node type as `master` in `/etc/xroad/conf.d/node.ini`:
+      ```bash
+      [node]
+      type=master
+      ```
+      Change the owner and group of the file to `xroad:xroad` if it is not already.
+8. Start the X-Road services.
```

Comment:
Maybe would be good to add "disable HTTP connection reuse (link to 3.5)" as a second-to-last step. It might be easy to miss 3.5 as chapter 3.3 ends with "The configuration is now complete. If you do not want to set up the health check service, continue to chapter 6 to verify the setup."